service/src/main/java/uk/gov/hmcts/reform/fpl/service/PopulateCaseService.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.enums.State;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.ResourceReader;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@SuppressWarnings("unchecked")
+public class PopulateCaseService {
+    private static final String FIXTURE_FILE_TEMPLATE = "e2e/fixtures/%s.json";
```

Comment:
that is not right, the packed code which is shipped to prod should not have depndenct to test folders
test files should not be there.
The test should pass json to endpoint

Reply:
That's a fair point, but using callback might be useful (for QAs and devs) to quickly create cases for testing without having to go through all manual steps. And if we're really concerned about storing these files in production code, they can be moved to e2e test directory and passed in the event as a Document. Will have a talk with QAs to get their opinion on it.